src/applications/facility-locator/actions/index.js

```diff
@@ -105,6 +105,14 @@ export const searchWithBounds = ({ bounds, facilityType, serviceType, page = 1 }
   const needsAddress = [LocationType.CC_PROVIDER, LocationType.ALL];
   // eslint-disable-next-line prettier/prettier
   return (dispatch) => {
+    dispatch({
+      type: SEARCH_STARTED,
+      payload: {
+        currentPage: page,
+        searchBoundsInProgress: true,
+      },
+    });
+
```

Comment:
This was a pain to hunt down. `searchWithBounds` is called during the VAMap's `UNSAFE_componentWillReceiveProps` when it detects the search query changed in order to refresh the location results list. A flag, `searchBoundsInProgress`, is flipped to prevent `searchWithBounds` from being called multiple times in `UNSAFE_componentWillReceiveProps`. This flag is set during the `SEARCH_STARTED` action, dispatched by a helper function, `fetchLocations`.

However, sometimes before `fetchLocations` is executed, we need to run `reverseGeocodeBox(bounds)`, which is async. During that async timeframe, the flag is not flipped, allowing multiple API requests to pass through.

Reply:
Doesn't this change create another gap? It seems like `SEARCH_STARTED` won't be dispatched now if we don't need to run `reverseGeocodeBox`. Or am I misunderstanding?

Reply:
Hm, maybe I'm looking at it wrong? I'm still relatively new to the FL code, so could missing something.

From what I could tell, `fetchLocations` is never directly executed - it's always just a helper for `searchWithBounds`. `searchWithBounds` dispatches `SEARCH_STARTED` before the if/else for `reverseGeocodeBox`, so it's always dispatched.

Reply:
Nope, you're right! I misread where the SEARCH_STARTED dispatch was being called, thought it was in reverseGeocodeBox